src/mmw/js/src/data_catalog/views.js

```diff
@@ -469,23 +464,156 @@ var ResultDetailsView = Marionette.ItemView.extend({
             placement: 'right',
             trigger: 'click',
         });
-        this.$('[data-toggle="table"]').bootstrapTable();
     },
 
+    closeDetails: function() {
+        this.model.collection.closeDetail();
+    }
+});
+
+var ResultDetailsCinergiView = ResultDetailsBaseView.extend({
+    template: resultDetailsCinergiTmpl,
+});
+
+var ResultDetailsHydroshareView = ResultDetailsBaseView.extend({
+    template: resultDetailsHydroshareTmpl,
+});
+
+var ResultDetailsCuahsiView = ResultDetailsBaseView.extend({
+    template: resultDetailsCuahsiTmpl,
+
     templateHelpers: function() {
         var id = this.model.get('id'),
-            location = id.substring(id.indexOf(':') + 1);
+            location = id.substring(id.indexOf(':') + 1),
+            fetching = this.model.get('fetching'),
+            error = this.model.get('error'),
+            last_date = this.model.get('end_date');
+
+        if (!fetching && !error) {
+            var variables = this.model.get('variables'),
+                last_dates = variables.map(function(v) {
+                        var values = v.get('values');
+
+                        if (values.length > 0) {
+                            return new Date(values.last().get('datetime'));
+                        } else {
+                            return new Date('01/01/1900');
+                        }
+                    });
+
+            last_dates.push(new Date(last_date));
+            last_date = Math.max.apply(null, last_dates);
+        }
 
         return {
             location: location,
+            last_date: last_date,
         };
     },
 
-    closeDetails: function() {
-        this.model.collection.closeDetail();
+    regions: {
+        valuesRegion: '#cuahsi-values-region',
+    },
+
+    ui: _.defaults({
+        chartButton: '#cuahsi-button-chart',
+        tableButton: '#cuahsi-button-table',
+    }, ResultDetailsBaseView.prototype.ui),
+
+    events: _.defaults({
+        'click @ui.chartButton': 'setChartMode',
+        'click @ui.tableButton': 'setTableMode',
+    }, ResultDetailsBaseView.prototype.events),
+
+    modelEvents: {
+        'change:fetching': 'render',
+        'change:mode': 'showValuesRegion',
+    },
+
+    initialize: function() {
+        var showValuesRegion = _.bind(this.showValuesRegion, this);
+
+        this.model.fetchCuahsiValues({
+            onEachSearchDone: showValuesRegion,
```

Comment:
Perhaps instead of reattaching a new instance of `CuahsiTableView` on each search done, we should simply `render` the currently attached one instead.